src/providers/iris/iris.ts

```diff
@@ -224,14 +226,17 @@ export class IrisProvider {
 
     // Get users according to params
     var returnObservable = startObservable
-      .flatMap(() => this.http.get<OncallUser[]>(`${this.irisInfo.baseUrl}${this.oncallApiPath}/users`, { params: new HttpParams({fromObject: params}) }));
+      .flatMap(() => this.http.get<OncallUser[]>(`${this.irisInfo.baseUrl}${this.oncallApiPath}/users?fields=name&fields=full_name`, { params: new HttpParams({fromObject: params}) }));
 
     return returnObservable
       .do(users => {
         this.clearOncallUsers();
         for (let i of users) {
           this.oncallUsers.push(i);
         }
+        this.oncallUsersLoaded = true;
+        this.refreshOncallUserCache();
```

Comment:
It looks like this function relies on values being populated in the IrisProvider's `oncallUsers` attribute. This means we need to wait on the promise returned by `refreshOncallUserCache` or use the stream value passed by the `http.get` observable